packages/react-native-payments/src/@standard/w3c/payment-details-init.ts

```diff
@@ -1,8 +1,11 @@
 import type { PaymentDetailsBase } from './payment-details-base';
 import type { PaymentItem } from './payment-item';
+import type { EnvironmentEnum } from '../../enum/environment.enum';
 
 // https://www.w3.org/TR/payment-request/#paymentdetailsinit-dictionary
 export interface PaymentDetailsInit extends PaymentDetailsBase {
+    // Android environment https://developers.google.com/android/reference/com/google/android/gms/wallet/Wallet.WalletOptions.Builder#setEnvironment(int)
+    environment: EnvironmentEnum;
```

Comment:
I do not understand why do you need this change if you are not using this field anywhere else in this PR? this is the level of the W3C implementation, it should not have any platform specific fields, all mapping to Android and iOS should be in corresponding classes.

Reply:
It is used not in JS side, but only in Native, [here](https://github.com/rnw-community/rnw-community/blob/master/packages/react-native-payments/android/src/main/java/com/payments/PaymentsModule.java#L150).
But if it's platform-specific as you're saying, I'll change the native code, so that environment is taken not from `paymentDetails`, but from `paymentMethodData`.

Reply:
have you seen the example usage for the android? https://github.com/rnw-community/rnw-community/blob/master/packages/react-native-payments-example/src/method-data/android-payment-method-data.ts it already accepts `environment` field.

Reply:
yes, I saw it. but it's incorrect, because environment is taken not from `methodData`, but from `paymentDetails` [here](https://github.com/rnw-community/rnw-community/blob/master/packages/react-native-payments/android/src/main/java/com/payments/PaymentsModule.java#L150).
So passing different environment to `methodData` doesn't change anything, it's always test environment.

Reply:
but I cannot change native code to take environment from `paymentMethodData` because `PaymentDataRequest` class from Google doesn't have `environment` as a field. Do you know what we should do?

Reply:
@vitalyiegorov maybe `environment` should be the third argument of Android's `show()` native method then?

Reply:
made changes according to your suggestions